internal/topology/variables/cluster_variable_validation.go

```diff
@@ -38,19 +38,19 @@ func ValidateClusterVariables(clusterVariables []clusterv1.ClusterVariable, clus
 	// * required variables from the ClusterClass exist on the Cluster.
 	// * all variables in the Cluster are defined in the ClusterClass.
 	allErrs = append(allErrs, validateRequiredClusterVariables(clusterVariablesMap, clusterClassVariablesMap, fldPath)...)
-	allErrs = append(allErrs, validateClusterVariablesDefined(clusterVariablesMap, clusterClassVariablesMap, fldPath)...)
+	allErrs = append(allErrs, validateClusterVariablesDefined(clusterVariables, clusterClassVariablesMap, fldPath)...)
```

Comment:
Why are we changing to an array here?

Reply:
With the array we're able to report the correct array index of undefined variables